app/Platform/Controllers/GameHashController.php

```diff
@@ -29,15 +34,100 @@ public function show(GameHash $gameHash): void
         dump($gameHash->toArray());
     }
 
+    public function manage(Game $game): View
+    {
+        $this->authorize('manage', $this->resourceClass());
+
+        $gameWithSortedHashes = Game::with(['hashes' => function ($query) {
+            $query->orderBy('Name');
+        }])->findOrFail($game->id);
+
+        return view('platform.manage-hashes-page', [
+            'game' => $gameWithSortedHashes,
+            'me' => Auth::user(),
+        ]);
+    }
+
     public function edit(GameHash $gameHash): void
     {
     }
 
-    public function update(Request $request, GameHash $gameHash): void
+    public function update(Request $request, GameHash $gameHash): JsonResponse
     {
+        $this->authorize('update', $this->resourceClass());
+
+        $input = $request->validate([
+            'name' => 'required|string',
+            'labels' => 'required|string',
+            'patch_url' => 'nullable|url|regex:/github\.com\/RetroAchievements\/RAPatches\/blob\/main\/.*\.zip$/i',
```

Comment:
Is `blob` the preferred path? @televandalist provided me a link that uses `raw` here: https://discord.com/channels/310192285306454017/1179551921263104112/1179821069461368842

Reply:
I think `blob` is indeed more accurate. Try navigating to: https://github.com/RetroAchievements/RAPatches/tree/main/NES/FDSConversion. Then right click on the file and copy the URL. The result should be:

`https://github.com/RetroAchievements/RAPatches/blob/main/NES/FDSConversion/14870-NewCluCluLand-FDSConversion.zip`

Reply:
But if you actually go to that URL: https://github.com/RetroAchievements/RAPatches/blob/main/NES/FDSConversion/14870-NewCluCluLand-FDSConversion.zip

It takes you to a github page, not the download itself. 

That has a link to the raw zip flile:
https://github.com/RetroAchievements/RAPatches/raw/main/NES/FDSConversion/14870-NewCluCluLand-FDSConversion.zip

Are we linking to the github page or the zip file directly?

Reply:
You're right - I should have actually checked where those links were going. Set to `raw` in latest.